security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -251,12 +272,13 @@ func (sf *SecretFetcher) scrtAdded(obj interface{}) {
 	rootCertResourceName := resourceName + IngressGatewaySdsCaSuffix
 	// If there is root cert secret with the same resource name, delete that secret now.
 	sf.secrets.Delete(rootCertResourceName)
-	if len(newRoot) > 0 {
+	if len(newCertAndKey.root) > 0 {
 		nsRoot := &model.SecretItem{
-			ResourceName: rootCertResourceName,
-			RootCert:     newRoot,
-			CreatedTime:  t,
-			Version:      t.String(),
+			ResourceName:              rootCertResourceName,
+			RootCert:                  newCertAndKey.root,
+			CertificateRevocationList: newCertAndKey.crl,
```

Comment:
do you want to check ```len(newCertAndKey.crl) > 0``` before setting CertificateRevocationList ?

Reply:
Good catch!  I'll fix this and the one below.